### PR TITLE
feat: Add single data value endpoint with JSON body [DHIS2-12689]

### PR DESCRIPTION
**Pull request by larshelge**
Adds a new endpoint for single data values with JSON payload support. The endpoint is available as `/api/dataValues` with `content-type: application/json`. Returns `201 Created`.

* Adds `DataValueDto`.
* Adds `DataValueCategoryDto`.

Sonar complains about one method but it is challenging to refactor and I don't want to make changes right before freeze.

### Testing

Added web API test for adding data value. Manual testing performed with Postman and data entry module.

### CODE REVIEW

File: dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/webdomain/datavalue/DataValueDto.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    private String period;
+
+    @JsonProperty
+    private String orgUnit;
+
+    @JsonProperty
+    private String categoryOptionCombo;
+
+    @JsonProperty
+    @JacksonXmlProperty
```
Comment (mortenoh) [inline dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/webdomain/datavalue/DataValueDto.java:60]: Why `@JacksonXmlProperty` on this one property?

Comment (larshelge) [reply dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/webdomain/datavalue/DataValueDto.java:60]: Removed now.

File: dhis-2/dhis-web-api-test/src/test/java/org/hisp/dhis/webapi/controller/DataValueControllerTest.java [hunk 2]
```diff
@@ -102,6 +102,22 @@ void testSetDataValuesFollowUp_Multi()
...
+        String body = String.format( "{" +
+            "'dataElement':'%s'," +
+            "'categoryOptionCombo':'%s'," +
+            "'period':'202201'," +
+            "'orgUnit':'%s'," +
+            "'value':'24'," +
+            "'comment':'OK'}",
+            dataElementId, categoryOptionComboId, orgUnitId );
+
+        HttpResponse response = POST( "/dataValues", body );
```
Comment (jbee) [inline dhis-2/dhis-web-api-test/src/test/java/org/hisp/dhis/webapi/controller/DataValueControllerTest.java:117]: Just so you know you could have done the parameter replacement as part of the `POST`.
There is an overloaded version where you point out non parameter by wrapping them.
```java
POST( "/dataValues", Body(template), dataElementId, categoryOptionComboId, orgUnitId );
```
The `Body` here is a static method in `WebClient` that would be imported for better readability. 
The `template` would be the string for your body payload just that you put `{<name>}` placeholder in there.
The name here is just informal, the parameters are substituted in order given.

**Review by jbee — APPROVED** [review]

File: dhis-2/dhis-web-api-test/src/test/java/org/hisp/dhis/webapi/controller/DataValueControllerTest.java (hunk 2 above)
Comment (larshelge) [reply dhis-2/dhis-web-api-test/src/test/java/org/hisp/dhis/webapi/controller/DataValueControllerTest.java:117]: Thanks, good to know.

**Pull request merged**